app/views/Project/Tagging/Export/NewExport/index.tsx

```diff
@@ -138,43 +128,75 @@ function NewExport(props: Props) {
     const {
         projectId,
     } = useParams<{ projectId: string }>();
+    const { state }: { state: ExportStateData | undefined } = useLocation();
     const history = useHistory();
     const [queryTitle, setQueryTitle] = useState<string | undefined>();
-    const [exportFileFormat, setExportFileFormat] = useState<ExportFormatEnum>('DOCX');
-    const [selectedLeads, setSelectedLeads] = useState<string[]>([]);
-    const [selectAll, setSelectAll] = useState<boolean>(true);
-
-    // advanced-options
-    const [reportShowLeadEntryId, setReportShowLeadEntryId] = useState<boolean>(true);
-    const [reportShowAssessmentData, setReportShowAssessmentData] = useState<boolean>(true);
-    const [reportShowEntryWidgetData, setReportShowEntryWidgetData] = useState<boolean>(true);
-    const [textWidgets, setTextWidgets] = useState<TreeSelectableWidget[]>([]);
+    const [
+        exportFileFormat,
+        setExportFileFormat,
+    ] = useState<ExportFormatEnum>(state?.format ?? 'DOCX');
+    const [
+        selectedLeads,
+        setSelectedLeads,
+    ] = useState<string[]>(state?.filters?.ids ?? []);
+    const [
+        selectAll,
+        setSelectAll,
+    ] = useState<boolean>(state?.filters?.excludeProvidedLeadsId ?? true);
+    const [
+        reportShowLeadEntryId,
+        setReportShowLeadEntryId,
+    ] = useState<boolean>(state?.extraOptions?.reportShowLeadEntryId ?? true);
+    const [
+        reportShowAssessmentData,
+        setReportShowAssessmentData,
+    ] = useState<boolean>(state?.extraOptions?.reportShowAssessmentData ?? true);
+    const [
+        reportShowEntryWidgetData,
+        setReportShowEntryWidgetData,
+    ] = useState<boolean>(state?.extraOptions?.reportShowEntryWidgetData ?? true);
+    const [
+        textWidgets,
+        setTextWidgets,
+    ] = useState<TreeSelectableWidget[]>([]);
     const [contextualWidgets, setContextualWidgets] = useState<TreeSelectableWidget[]>([]);
     const [reportStructure, setReportStructure] = useState<Node[]>([]);
     const [includeSubSector, setIncludeSubSector] = useState<boolean>(false);
     const [reportStructureVariant, setReportStructureVariant] = useState<string>(SECTOR_FIRST);
-    const [excelDecoupled, setExcelDecoupled] = useState<boolean>(false);
+    const [
+        excelDecoupled,
+        setExcelDecoupled,
+    ] = useState<boolean>(state?.extraOptions?.excelDecoupled ?? false);
 
     const {
         value: sourcesFilterValue,
-        setFieldValue: setSourcesFilterValue,
+        setFieldValue: setSourcesFilterFieldValue,
+        setValue: setSourcesFilter,
         resetValue: clearSourcesFilterValue,
         pristine,
         validate,
         setError,
         setPristine,
     } = useFilterState();
 
+    useEffect(() => {
+        if (state?.filters) {
+            setSourcesFilter(transformRawFiltersToFormValues(state?.filters));
+        }
+    }, [setSourcesFilter, state?.filters]);
```

Comment:
Does this render only on the first render?

Reply:
yes